wdl-lint/src/rules/misplaced_lint_directive.rs

```diff
@@ -0,0 +1,133 @@
+    exceptable_nodes: &[SyntaxKind],
+) -> Diagnostic {
+    let locations = exceptable_nodes
+        .iter()
+        .map(|node| node.describe())
+        .collect::<Vec<_>>()
+        .join(", ");
+
+    Diagnostic::note(format!(
+        "lint directive `{id}` has no effect above: {elem}",
+        elem = wrong_element.kind().describe()
+    ))
+    .with_rule(ID)
+    .with_label("cannot make an exception for this rule", span)
+    .with_label(
+        "invalid element for this lint directive",
+        wrong_element.text_range().to_span(),
+    )
+    .with_fix(format!(
+        "valid locations for this directive are above: {locations}"
+    ))
+}
+
+/// Detects unknown rules within lint directives.
+#[derive(Default, Debug, Clone, Copy)]
+pub struct MisplacedLintDirective;
+
+impl Rule for MisplacedLintDirective {
+    fn id(&self) -> &'static str {
+        ID
+    }
+
+    fn description(&self) -> &'static str {
+        "Flags misplaced lint directives which will have no effect."
+    }
+
+    fn explanation(&self) -> &'static str {
+        "When writing WDL, lint directives are used to suppress certain rules. If a lint directive \
+         is misplaced, it will have no effect. This rule flags misplaced lint directives to ensure \
+         they are in the correct location."
+    }
+
+    fn tags(&self) -> TagSet {
+        TagSet::new(&[Tag::Clarity, Tag::Correctness])
+    }
+
+    fn exceptable_nodes(&self) -> Option<Vec<wdl_ast::SyntaxKind>> {
+        Some(vec![SyntaxKind::VersionStatementNode])
+    }
+}
+
+impl Visitor for MisplacedLintDirective {
+    type State = Diagnostics;
+
+    fn document(&mut self, _: &mut Self::State, _: VisitReason, _: &Document, _: SupportedVersion) {
+        // This is intentionally empty, as this rule has no state.
+    }
+
+    fn comment(&mut self, state: &mut Self::State, comment: &Comment) {
+        if let Some(ids) = comment.as_str().strip_prefix(EXCEPT_COMMENT_PREFIX) {
+            let start: usize = comment.span().start();
+            let mut offset = EXCEPT_COMMENT_PREFIX.len();
+
+            let excepted_element = comment
+                .syntax()
+                .siblings_with_tokens(rowan::Direction::Next)
+                .find_map(|s| {
+                    if s.kind() == SyntaxKind::Whitespace || s.kind() == SyntaxKind::Comment {
+                        None
+                    } else {
+                        Some(s)
+                    }
+                });
+
+            for id in ids.split(',') {
+                // First trim the start so we can determine how much whitespace was removed
+                let trimmed_start = id.trim_start();
+                // Next trim the end
+                let trimmed: &str = trimmed_start.trim_end();
+
+                // Update the offset to account for the whitespace that was removed
+                offset += id.len() - trimmed.len();
+
+                if let Some(rule) = rules().iter().find(|r| r.id() == trimmed) {
```

Comment:
`rules` is an expensive call, unfortunately.

At worst, we should cache the call on construction of `MisplacedLintDirective` (it shouldn't go in `document` as the lint rule list never changes).

At best, perhaps we lazy construct (using `LazyLock`) a static `HashMap<&'static str, Option<&'static [SyntaxKind]>>` which gets populated from a single call to `rules`. When we can easily go from rule name to its exceptable nodes.